### PR TITLE
chore: migrate account references to async/await

### PR DESCRIPTION
**Pull request by jniles**
This commit migrates accounts references to asnyc/await format.

### CODE REVIEW

File: server/controllers/finance/accounts/references.js [hunk 1]
```diff
@@ -257,31 +261,27 @@ function getValue(req, res, next) {
...
-      return glb;
-    });
+  const reference = await db.one(sql, id);
+
+  const [accounts, accountsException] = await Promise.all([
+    db.exec(sqlItems, [id]),
+    db.exec(sqlExceptItems, [id]),
+  ]);
+
+  Object.assign(reference, { accounts, accountsException });
```
Comment (mbayopanda) [inline server/controllers/finance/accounts/references.js]: `accounts` and `accountsException` must be arrays of integer, not arrays of objects `{ account_id : ... }`.

we must have something like : 
```js
reference.accounts = accounts.map(item => item.account_id);
reference.accountsException = accountsException.map(item => item.account_id);
```

**Comment by jniles** [comment]
@mbayopanda updates landed.

File: server/controllers/finance/accounts/references.js [hunk 2]
```diff
@@ -257,31 +261,31 @@ function getValue(req, res, next) {
...
+  const reference = await db.one(sql, id);
+
+  const [accounts, accountsException] = await Promise.all([
+    db.exec(sqlItems, [id]),
+    db.exec(sqlExceptItems, [id]),
+  ]);
+
+  Object.assign(reference, {
+    accounts : _.flatMap(accounts, 'account_id'),
+    accountsException : _.flatMap(accountsException, 'account_id'),
```
Comment (mbayopanda) [inline server/controllers/finance/accounts/references.js:285]: 👍

**Review by mbayopanda — APPROVED** [review]
LGTM 👍 

bors r+

**Pull request merged**